### ISSUE TITLE
wrong Cardboard profiles - a problem for VR Quality

### ISSUE THREAD

**Issue by noemis84** [googlevr/gvr-unity-sdk#449]
I'm not sure, if this is the right place for such a discussion (which will hopefully start), but because we use the googlevr unity sdk for all our cardboard apps, it feels right.

## The PROBLEM
We have a variety of cardboards. But everytime we get a new cardboard or buy one, the experience is different.

I claim that there are many **cardboards produced with completly wrong profiles** and the normal user has no chance to get a good VR experience. They will even made wrong conclusions:
- that they got the wrong phone
- the app is bad
- his eyes are not made for VR

## The TEST
To keep it simple, we should concentrate on cardboard V2. Here are my results for these four devices...

![overview_small](https://cloud.githubusercontent.com/assets/13518143/22422486/dfe7f9b4-e6ec-11e6-8745-eecc8c5546a6.jpg)

What I've done?
1. scan cardboard profile with something like qr code scanner
2. copy short url and send it for example via email to you
3. open link in browser (PC) and copy the profile (everything behind ?p=)
4. on https://wwgc.firebaseapp.com/ you can press "Save or Load Viewer Parametes" and use this profile to get all values

I publish the details in this order (like the order, when you create a profile):
Name
Profilecode
Screen to Lens / Inter-lens / Screen vertical alignment / Tray to lens-center distance / k1 / k2

1. Unofficial Cardboard (v2.0)
CgASAB0xCCw9JY_CdT0qEAAASEIAAEhCAABIQgAASEJYADUpXA89OggAAAAAAAAAAFAAYAA
45mm / 62,0mm / Bottom / 40mm / 0 / 0

2. ZAAK ZURICH BERLIN (Cardboard V2)
ChJaQUFLIFpVUklDSCBCRVJMSU4SDENhcmRib2FyZCBWMh0xCCw9JbbzfT0qEAAASEIAAEhCAABIQgAASEJYADUpXA89Oghcj0I-j8L1PlAAYAM
42mm / 62mm / Bottom / 35mm / 0.190 / 0.480

3. IHUAQI Technology Ltd. (Cardboard VR)
ChZJSFVBUUkgVGVjaG5vbG9neSBMdGQuEgxDYXJkYm9hcmQgVlIdzcxMPSWWQ4s9KhAAAEhCAABIQgAASEIAAEhCWAA1KVwPPToICtcjPI_C9TxQAGAD
50mm / 68mm / Bottom / 35mm / 0.010 /0.030

4. Shoogee GmbH & Co. KG (Cardboard 2) = Dive Cardboard
ChVTaG9vZ2VlIEdtYkggJiBDby4gS0cSC0NhcmRib2FyZCAyHTEILD0ltvN9PSoQAABIQgAASEIAAEhCAABIQlgANbx0Ez06COxRuD4AAAAAUABgAw
42mm / 62mm / Bottom / 36mm / 0.360 / 0.000

## RESULT
Even if ALL cardboards have been bought as an V2 and they all look very similar from outside, two of them are not useable in VR. Number 2. and 4. have the best result (when see through). That's what everybody should get from a working cardboard profile: at least visible lens distortion. **The most problematic number seem to be the "Distortion coefficients"** We bought 3. just some weeks ago over amazon. I got also another cardboard at home which is as new and bad as 3.

Here a **comparison of the left eye**.... notice, that _the white ui circle on TV is always slighlty above the tv logo_. And now compare the images... horrible!
![alltypeslefteye_small](https://cloud.githubusercontent.com/assets/13518143/22422583/8197ddba-e6ed-11e6-9a6d-d99d2f147b55.jpg)

## What should be done?
- as long as cardboard are common vr device, we should complain/inform about wrong profiles.
- maybe google can check some profiles,...
- what do you think?

**Comment by noemis84** [comment]
Dear manufacturers: READ THE FU***** manual:
https://support.google.com/cardboard/manufacturers/answer/6324808?hl=en

**Comment by noemis84** [comment]
Now I tested my cardboard at home:

### 5. BrizTech Ltd (Google Cardboard 2.0)
CgxCcml6VGVjaCBMdGQSFEdvb2dsZSBDYXJkYm9hcmQgMi4wHexROD0lbxKDPSoQAABIQgAASEIAAEhCAABIQlgANXe-Hz06CAAAAAAAAAAAUABgAw
45mm / 64mm / Bottom / 39mm / 0.000 / 0.000

the profile editor marks the k1 and k2 entries with the red words: "...coefficient should not be left set to 0.00."

And also this cardboard look completly WRONG. Current statistic says: 3 of 5 cardboards aka 60% using a wrong profile... wow!

**Comment by noemis84** [comment]
To understand, how such a profile is generated and also prevent most of my cardboards going to trash I decided to **configure them on my own**. Even, this is not meant to be done neither by the developers (like me) nor the user, I still would recomend it to everyone who is interested in VR and in improving things ;)

I noticed, that it does not take much time (shame on you manufacturers) and the cardboard viewing quality with your specific device improves... because the viewing profile in general "tells" the smartphone details about the specs from the cardboard, but as I had to learn: they are mostly not correct and those adjustment are depentdend from YOU... because your way of holding the cardboard, your eye distance, if you wear glasses,etc... too much differences from person to person.

To generate my "personal viewing profile" I used this guide:
https://support.google.com/cardboard/manufacturers/checklist/6322188

Many things to read and check but at least I took me about 20 - 30min to make this qrcode:
![qr_viewer_profile_briztech_small](https://cloud.githubusercontent.com/assets/13518143/22443291/afdac254-e73e-11e6-80be-eb0c3da2a77a.jpg)
**important note:** This is my personal adjusted profile for the BrizTech Cardboard 2.0 used with a MotoZ! I was very suprised about some values (e.g. smaller eye distance than expected), but the most important settings are: **k1 and k2 - just don't leave them at 0!**

## Result
1. The profile works about 300% better than the printed on the cardboard itself.
2. I'm not sure, if I'm confused after all this testing stuff, but it was **maybe the most comfort cardboard experience I ever had**... the middle of the screen, eye distance, etc. works just great. I really liked the "Tray to lens-center" adjustment, because very often I have the feeling that the center of the screen is too low or too high... not the case with my **pvp** (personal viewing profile ;)

### To sum up: I'm still very interested in some of your experiences with cardboard profiles, data of your devices and maybe you try also creating your own profile?

**Comment by miraleung** [comment]
Thank you for your feedback, we've passed it on to the relevant teams for further investigation.

**Comment by Rubicsys** [comment]
That's interesting.  I tried calibrating my S6 (as recommended on the initial page of the checklist sequence) and it comes up with the error "Mobile device display properties unknown....."

**Comment by noemis84** [comment]
Yes, I had also this "error" with my Moto Z, but could solve it by typing in my ppi value...

**Comment by Rubicsys** [comment]
Thanks.  I assumed from earlier words that this message means it is not a suitable phone for calibrating this way. I'll try again.
